### ISSUE TITLE
Custom angular component will cause app crash on iOS if wrapped in FlexboxLayout.

### ISSUE THREAD

**Issue by jamieathans** [NativeScript/nativescript-angular#629]
Given the following angular components, **AppComponent** and **TestComponent**:

```
import { Component } from "@angular/core";

@Component({
    selector: "my-app",
    template: `
        <FlexboxLayout>
            <my-test></my-test>
        </FlexboxLayout>
    `
})
export class AppComponent {
}

@Component({
    selector: "my-test",
    template: `
        <Label text="Test"></Label>
    `
})
export class TestComponent {
}
```

This above code will cause a crash on iOS with the following error:

```
***** Fatal JavaScript exception - application has been terminated. *****
Native stack trace:
1   0x100517764 NativeScript::FFICallback<NativeScript::ObjCMethodCallback>::ffiClosureCallback(ffi_cif*, void*, void**, void*)
2   0x100da12cc ffi_closure_SYSV_inner
3   0x100da41b4 .Ldo_closure
4   0x188ec7a80 <redacted>
5   0x1863759d8 <redacted>
6   0x18636a4cc <redacted>
7   0x18636a38c <redacted>
8   0x1862e73e0 <redacted>
9   0x18630ea68 <redacted>
10  0x18630f488 <redacted>
11  0x182fc20c0 <redacted>
12  0x182fbfcf0 <redacted>
13  0x182fc0180 <redacted>
14  0x182eee2b8 CFRunLoopRunSpecific
15  0x1849a2198 GSEventRunModal
16  0x188f357fc <redacted>
17  0x188f30534 UIApplicationMain
18  0x100da4044 ffi_call_SYSV
19  0x100da0e0c ffi_call_int
20  0x100da0924 ffi_call
21  0x1004d94fc NativeScript::FFICall::call(JSC::ExecState*)
22  0x100b0b888 JSC::LLInt::setUpCall(JSC::ExecState*, JSC::Instruction*, JSC::CodeSpecializationKind, JSC::JSValue, JSC::LLIntCallLinkInfo*)
23  0x100b13f08 llint_entry
24  0x100b13f18 llint_entry
25  0x100b13f18 llint_entry
26  0x100b13f18 llint_entry
27  0x100b0d7f8 vmEntryToJavaScript
28  0x100aad294 JSC::JITCode::execute(JSC::VM*, JSC::ProtoCallFrame*)
29  0x100a8a230 JSC::Interpreter::executeCall(JSC::ExecState*, JSC::JSObject*, JSC::CallType, JSC::CallData const&, JSC::JSValue, JSC::ArgList con
st&)
30  0x100be9e64 JSC::call(JSC::ExecState*, JSC::JSValue, JSC::CallType, JSC::CallData const&, JSC::JSValue, JSC::ArgList const&, WTF::NakedPtr<JSC
::Exception>&)
31  0x1004e9ce4 NativeScript::GlobalObject::moduleLoaderEvaluate(JSC::JSGlobalObject*, JSC::ExecState*, JSC::JSModuleLoader*, JSC::JSValue, JSC::J
SValue, JSC::JSValue)
JavaScript stack trace:
1   measure@file:///app/tns_modules/ui/core/view.js:96:124
2   @file:///app/tns_modules/ui/layouts/flexbox-layout/flexbox-layout.js:287:30
3   _measureHorizontal@file:///app/tns_modules/ui/layouts/flexbox-layout/flexbox-layout.js:317:11
4   onMeasure@file:///app/tns_modules/ui/layouts/flexbox-layout/flexbox-layout.js:137:40
5   measure@file:///app/tns_modules/ui/core/view.js:93:27
6   measureChild@file:///app/tns_modules/ui/core/view-common.js:977:26
7   onMeasure@file:///app/tns_modules/ui/page/page.js:374:46
8   measure@file:///app/tns_modules/ui/core/view.js:93:27
9   measureChild@file:///app/tns_modules/ui/core/view-common.js:977:26
10  measurePage@file:///app/tns_modules/ui/frame/frame.js:234:40
11  onMeasure@file:///app/tns_modules/ui/frame/frame.js:216:38
12  measure@file:///app/tns_modules/ui/core/view.js:93:27
13  _layoutRootView@file:///app/tns_modules/ui/utils.js:42:25
14  layoutSubviews@file:///app/tns_modules/application/application.js:41:40
15  UIApplicationMain@[native code]
16  start@file:///app/tns_modules/application/application.js:251:30
17  bootstrapApp@file:///app/tns_modules/nativescript-angular/platform-common.js:67:28
18  bootstrapModule@file:///app/tns_modules/nativescript-angular/platform-common.js:55:26
19  anonymous@file:///app/main.js:5:57
20  evaluate@[native code]
21  moduleEvaluation@[native code]
22  @[native code]
23  promiseReactionJob@[native code]
JavaScript error:
file:///app/tns_modules/ui/core/view.js:96:124: JS ERROR Error: onMeasure() did not set the measured dimension by calling setMeasuredDimension() P
roxyViewContainer(6)
```

A workaround is to wrap the custom component at the call-site in a **StackLayout**.

```
@Component({
    selector: "my-app",
    template: `
        <FlexboxLayout>
            <StackLayout><my-test></my-test></StackLayout>
        </FlexboxLayout>
    `
})
export class AppComponent {
}
```

This was tested on a new project using:

`tns create tns-test-ng --ng`

tns info:

```
All NativeScript components versions information
┌──────────────────┬─────────────────┬────────────────┬─────────────┐
│ Component        │ Current version │ Latest version │ Information │
│ nativescript     │ 2.4.2           │ 2.4.2          │ Up to date  │
│ tns-core-modules │ 2.4.4           │ 2.4.4          │ Up to date  │
│ tns-android      │ 2.4.1           │ 2.4.1          │ Up to date  │
│ tns-ios          │ 2.4.0           │ 2.4.0          │ Up to date  │
└──────────────────┴─────────────────┴────────────────┴─────────────┘
```

**Comment by tsonevn** [comment]
Hi @jamieathans,

I reviewed your case and confirm that this is a real problem, while using custom angular component inside the `FlexboxLayout`. This issue has been reproduced only on `iOS`.

For further info, you could keep track this issue.

Regards,
@tsonevn

**Comment by NathanWalker** [comment]
Just curious and following up here. I was thinking this fix may coincide with the platform 2.5 release?

**Comment by vchimev** [comment]
Fixed in **tns-core-modules@2.5.2**.
